### ISSUE TITLE
`load_schema` does not account for dictionary settings

### ISSUE THREAD

**Issue by Sophon96** [ThatOtherAndrew/ThatOtherBot#131]
### Checklist

- [x] I have been able to successfully and reliably reproduce the issue.
- [x] I confirm this issue is present as of the latest commit on the `main` branch.
- [x] I have searched for a similar issue in this repository and didn't find anything relevant.

### Description

Breadcord does not account for dictionary settings and fails when loading a module that has dictionary settings in its `settings_schema.toml`.

### Reproduction

1. Have a module with a dictionary setting in `settings_schema.toml`

### Expected behavior

The bot correctly loads the setting as a dictionary.

### Attachments

(No attachments)

### Extra information

The reason for this error is that Breadcord expects all Python `dict` objects returned by tomlkit to be TOML tables. This behavior was introduced in 41073865546e745a236592a973f2789abcf7fc61 and is incorrect. TOML specification supports dictionary types in TOML files.

**Comment by Sophon96** [comment]
Correction, dicts **are** tables. The code does not correctly account for tables.

**Comment by Fripe070** [comment]
Could you provide an example `settings_schema.toml` file and the expected output?

**Comment by Sophon96** [comment]
> Could you provide an example `settings_schema.toml` file and the expected output?

Sure!
```toml
outer = { inner = 42 }
```
A module with such a `settings_schema.toml` will fail to load.

**Comment by Sophon96** [comment]
> Could you provide an example `settings_schema.toml` file and the expected output?

The expected output would be a setting group `outer` in `self.settings` and a setting `inner` in the setting group `outer`.

**Comment by Fripe070** [comment]
Right, you are talking about [inline tables](https://toml.io/en/v1.0.0#inline-table)

**Comment by Sophon96** [comment]
Yep!
